spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/TensorflowToBigDL.scala

```diff
@@ -0,0 +1,1091 @@
+      if (classTag[T] == classTag[Float]) {
+        if (tfTensor.getDtype == DataType.DT_FLOAT) {
+          return Tensor[Float](T(tfTensor.getFloatVal(0))).asInstanceOf[Tensor[T]]
+        }
+
+        if (tfTensor.getDtype == DataType.DT_INT32) {
+          return Tensor[Float](T(tfTensor.getIntVal(0).toFloat)).asInstanceOf[Tensor[T]]
+        }
+
+        throw new IllegalArgumentException("Can not convert double to float")
+      } else if (classTag[T] == classTag[Double]) {
+        if (tfTensor.getDtype == DataType.DT_DOUBLE) {
+          return Tensor[Float](T(tfTensor.getDoubleVal(0))).asInstanceOf[Tensor[T]]
+        }
+
+        if (tfTensor.getDtype == DataType.DT_FLOAT) {
+          return Tensor[Float](T(tfTensor.getFloatVal(0).toDouble)).asInstanceOf[Tensor[T]]
+        }
+
+        if (tfTensor.getDtype == DataType.DT_INT32) {
+          return Tensor[Float](T(tfTensor.getIntVal(0).toDouble)).asInstanceOf[Tensor[T]]
+        }
+      }
+    }
+
+    val buffer = ByteBuffer.wrap(tfTensor.getTensorContent.toByteArray)
+    buffer.order(endian)
+
+    if (classTag[T] == classTag[Float]) {
+      if (tfTensor.getDtype == DataType.DT_FLOAT) {
+        val params = buffer.asFloatBuffer
+        val tmp = new Array[Float](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else if (tfTensor.getDtype == DataType.DT_INT32) {
+        val params = buffer.asIntBuffer
+        val tmp = new Array[Float](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else {
+        throw new IllegalArgumentException("Can not convert double to float")
+      }
+    } else if (classTag[T] == classTag[Double]) {
+      if (tfTensor.getDtype == DataType.DT_FLOAT) {
+        val params = buffer.asFloatBuffer
+        val tmp = new Array[Double](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else if (tfTensor.getDtype == DataType.DT_INT32) {
+        val params = buffer.asIntBuffer
+        val tmp = new Array[Double](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
+      } else if (tfTensor.getDtype == DataType.DT_DOUBLE) {
+        val params = buffer.asDoubleBuffer()
+        val tmp = new Array[Double](params.capacity())
+        var j = 0
+        while (j < params.capacity()) {
+          tmp(j) = params.get(j)
+          j += 1
+        }
+        Tensor(Storage(tmp), 1, shape).asInstanceOf[Tensor[T]]
```

Comment:
It seems a little strange, there are five similar code snippets. Does this have much higher performance?